hydra/utils.py

```diff
@@ -50,7 +50,10 @@ def instantiate(config: DictConfig, *args: Any, **kwargs: Any) -> Any:
 
     assert config is not None, "Input config is None"
     # copy config to avoid mutating it when merging with kwargs
-    config = copy.deepcopy(config)
+    config_copy = copy.deepcopy(config)
+    if config._get_parent() is not None:
```

Comment:
you can set a None parent. did you ran into issues?

Reply:
Ohh I did not try. I can add a separate testcase for that. I added the check to be on the safe side.

Reply:
Yes. I don't think you can even create the test set. params always has a parent.

Reply:
How about this: 
```@pytest.mark.parametrize(  # type: ignore
    "input_conf, expected",
    [
        (
            {
                    "main": {
                        "class": "tests.test_utils.Bar",
                        "params": {"a": 10, "b": 20, "c": "${all_params.aux.c}"},
                    },
                    "aux": {"c": 30},
                },
            Bar(10, 20, 30, 40),
        ),
    ],
)
def test_string_iterpolation_during_deepcopy(
    input_conf: Dict[str, Any], expected: Any
) -> None:
    # Check if the instantiate method maintains the parent when making a deepcopy
    conf = OmegaConf.create(input_conf)
    orig = copy.deepcopy(conf)
    assert isinstance(conf, DictConfig)
    obj = utils.instantiate(conf.main, d=40)
    assert orig == conf
    assert obj == expected
```

Reply:
Ohh and I can set a None parent :)

Reply:
Removed the none check